autogpt_platform/backend/schema.prisma

```diff
@@ -277,3 +278,41 @@ model UserBlockCredit {
 
   @@id(name: "creditTransactionIdentifier", [transactionKey, userId])
 }
+
+enum APIKeyPermission {
+  EXECUTE_GRAPH // Can execute agent graphs
+  READ_GRAPH // Can get graph versions and details
+  EXECUTE_BLOCK // Can execute individual blocks
+  READ_BLOCK // Can get block information
+}
+
+model APIKey {
+  id          String             @id @default(uuid())
+  name        String
+  prefix      String             @unique // First 8 chars for identification
```

Comment:
what's `prefix`?

Reply:
OH I get it. Maybe it's my english but I find `prefix` a slightly confusing term here. E.g. I'd expect the prefix to be something like the `sk-proj-` in OpenAI API keys or `sk-ant-api03-` for Anthropic.

Maybe call it `preview`? Also, I don't think we can require this column to be unique as we're only using a small part of the actual key.

Reply:
I hear your point about prefix potentially being confusing, I think preview might actually be less precise here. The term preview suggests we're showing a representative sample of the key, which could be any part of it (middle, end, etc). But we're specifically storing the beginning/leftmost portion, which is technically a prefix. 


On uniqueness, sure I can remove that

Reply:
You could also add a postfix too if you want

Reply:
>  You could also add a postfix too if you want

what usecase are you thinking for postfix as well as prefix?

Reply:
Showing apikey preview as form agpt-dhehbrkwowj*********ahhekso

See also this when choosing format
https://docs.github.com/en/code-security/secret-scanning/secret-scanning-partnership-program/secret-scanning-partner-program

Reply:
Added postfix